File: MaxRent.API/Models/OrderItem.cs
using System;
using System.Collections.Generic;

namespace MaxRent.API.Models
{
    public class OrderItem
    {
        public int Id { get; set; }
        public DateTime DateFrom { get; set; }
        public DateTime DateTo { get; set; }
        public double TotalPrice { get; set; }
        public ICollection<OrderItemAsset> OrderItemAssets { get; set; }
        public Order Order { get; set; }
        public int OrderId { get; set; }
    }
}

File: maxrent-SPA/src/app/_models/order-item.ts
import { Product } from './product';

export class OrderItem {
    product: Product;
    dateFrom: Date;
    dateTo: Date;
    totalPrice: number;
    assetsCount: number;
    getTotalDays(): number {
         return (this.dateTo.getDate() - this.dateFrom.getDate() + 1 );
    }
}
